bpf-loader-rs/bpf-loader-lib/src/skeleton/preload/attach.rs

```diff
@@ -30,10 +32,36 @@ impl Drop for AttachLink {
                     error!("Failed to destroy tc hook: {}", err);
                 }
             }
+            AttachLink::XDPAttach(ifindex, flags, opts) => {
+                let err = unsafe { bpf_xdp_detach(*ifindex, *flags, &**opts) };
+                if err != 0 {
+                    error!("Failed to detach xdp hook");
+                }
+            }
         }
     }
 }
 
+pub(crate) fn attach_xdp(program: &Program, meta: &ProgMeta) -> Result<AttachLink> {
```

Comment:
libbpf-rs provides APIs to attach XDP. https://docs.rs/libbpf-rs/latest/libbpf_rs/struct.Program.html#method.attach_xdp . is this suitable for us?

Reply:
It's not suitable